signac/synced_collections/backends/collection_json.py

```diff
@@ -278,24 +277,26 @@ def _lock_id(self):
 
 
 # These are the common protected keys used by all JSONDict types.
-_JSONDICT_PROTECTED_KEYS = (
-    # These are all protected keys that are inherited from data type classes.
-    "_data",
-    "_name",
-    "_suspend_sync_",
-    "_load",
-    "_sync",
-    "_root",
-    "_validators",
-    "_all_validators",
-    "_load_and_save",
-    "_suspend_sync",
-    "_supports_threading",
-    "_LoadSaveType",
-    "registry",
-    # These keys are specific to the JSON backend.
-    "_filename",
-    "_write_concern",
+_JSONDICT_PROTECTED_KEYS = frozenset(
+    [
+        # These are all protected keys that are inherited from data type classes.
+        "_data",
+        "_name",
+        "_suspend_sync_",
+        "_load",
+        "_sync",
+        "_root",
+        "_validators",
+        "_all_validators",
+        "_load_and_save",
+        "_suspend_sync",
+        "_supports_threading",
+        "_LoadSaveType",
+        "registry",
+        # These keys are specific to the JSON backend.
+        "_filename",
+        "_write_concern",
+    ]
 )
```

Comment:
Good call on the mutability. I don't think the benchmarks is super relevant since this will only happen one time ever (when the class is defined, rather than when instances are created), but the change is a good one for clarity.